feat(sdk): add kmp platform (#6452)
getsentry/sentry-docs: src/platforms/kotlin-multiplatform/initialization-strategies.mdx

fun initializeSentry(context: Context?) {
  Sentry.init(context) {
    it.dsn = "___PUBLIC_DSN___"
  }
}
```

### Android

```kotlin {filename:MainActivity.kt}
import your.kmp.app.initializeSentry

class YourApplication : Application() {
  override fun onCreate() {
    super.onCreate()
      // Make sure to add the context!
      initializeSentry(this)
   }
}
```

### Cocoa/Apple

```swift {filename:AppDelegate.swift}
import shared

func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
) -> Bool {
    AppSetupKt.initializeSentry(context = nil)
    return true
}
```

## Platform specific initializers

Platform specific initializers allow for customization of the SDK's configuration options on a per-platform basis. This approach can be particularly useful when your SDK requires platform specific functionality or performance optimizations.

With platform-specific initializers, you can initialize the SDK directly in the target platform's codebase. This approach gives you the flexibility to fine-tune the SDK's behavior to meet the unique needs of each platform.

> ### Android
> ```kotlin {filename:MainActivity.kt}

Reviewer: ```suggestion
### Android

```kotlin {filename:MainActivity.kt}
```

There are more cases that require jumping a line.

---
title: Initialization Strategies
sidebar_order: 0
description: "Different strategies how to initialize the Kotlin Multiplatform SDK."
---

When it comes to initializing a Kotlin Multiplatform SDK, there are three strategies to consider:
 - [Shared initializer](#shared-initializer)
 - [Platform specific initializers](#platform-specific-initializers)
 - [Hybrid approach](#hybrid-approach)

## Shared initializer

> The shared initializer approach involves initializing the SDK in your shared codebase, using the same configuration options for all platforms. This approach is ideal for projects that prioritize consistency across platforms and do not require platform-specific customizations.
> 
> Using a shared initializer provides a single source of truth for your SDK's configuration options. This can simplify maintenance and debugging efforts, as you only need to update one codebase for all platforms.
> 
> Inside a Kotlin file in your commonMain e.g. `AppSetup.kt` or however you want to call it and create a function that will initialize the SDK.
> Afterwards you can call it in an early lifecycle stage in your platforms.

Reviewer 2: ```suggestion
The shared initializer approach involves initializing the SDK in your shared codebase, using the same configuration options for all platforms. This approach is ideal for projects that prioritize consistency across platforms and don't require platform-specific customizations.

Using a shared initializer provides a single source of truth for your SDK's configuration options. This can simplify maintenance and debugging efforts, as you only need to update one codebase for all platforms.

To initialize the SDK, create a Kotlin file in your commonMain e.g. `AppSetup.kt`, (or whatever you decide to call it), and create an initialization function. You'll then be able to call it in an early lifecycle stage in your platforms.
```

Not sure I understood this correctly, but this section read like there was missing information originally. (What do we want the user to do inside their Kotlin file?)

Author: The action that the user needs to do in their Kotlin file (AppSetup.kt) refers to the code bit that is shown immediately after this text.

    return true
}
```

## Platform specific initializers

Platform specific initializers allow for customization of the SDK's configuration options on a per-platform basis. This approach can be particularly useful when your SDK requires platform specific functionality or performance optimizations.

With platform-specific initializers, you can initialize the SDK directly in the target platform's codebase. This approach gives you the flexibility to fine-tune the SDK's behavior to meet the unique needs of each platform.

### Android

```kotlin {filename:MainActivity.kt}
import io.sentry.kotlin.multiplatform.Sentry

class YourApplication : Application() {
  override fun onCreate() {
    super.onCreate()
      // Make sure to add the context!
      Sentry.init(this) {
        it.dsn = "___PUBLIC_DSN___"
      }
   }
}
```

### Cocoa/Apple

```swift {filename:AppDelegate.swift}
import shared

func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
) -> Bool {
    Sentry.shared.doInit() { options in
      options.dsn = "___PUBLIC_DSN___"
    }
    return true
}
```

## Hybrid approach

> It is also possible to mix the two initialization strategies by creating custom sourcesets that only target specific platforms. This allows you to use a shared initializer for some platforms while utilizing platform-specific initializers for others.
> 
> For example, you may choose to use a shared initializer for Android and iOS platforms while using a platform specific initializer for the watchOS and tvOS platform. This approach provides a balance between consistency and customization.

Reviewer 2: ```suggestion
It's also possible to mix the two initialization strategies by creating custom sourceSets that only target specific platforms. This allows you to use a shared initializer for some platforms while utilizing platform-specific initializers for others.

For example, you may choose to use a shared initializer for Android and iOS platforms, while using a platform-specific initializer for the watchOS and tvOS platform. This approach provides a balance between consistency and customization.
```
